# Test Cricket Data Analysis
<br>goal of this notebook is to analyze test cricket data in given dataset

## Dataset sourse- https://stats.espncricinfo.com/ci/content/records/93276.html

### Import required libraries

In [1]:
import numpy as np
import pandas as pd

### Reading xlsx file

In [3]:
df = pd.read_excel("test_cricket.xlsx", engine='openpyxl' , sheet_name='wickets')

### First 10 rows of dataset

In [4]:
display(df.head(10))

Player       Span  Mat  Inns  Balls   Runs  Wkts  \
0  M Muralitharan (ICC/SL)  1992-2010  133   230  44039  18180   800   
1           SK Warne (AUS)  1992-2007  145   273  40705  17995   708   
2         A Kumble (INDIA)  1990-2008  132   236  40850  18355   619   
3        JM Anderson (ENG)  2003-2021  162   301  34791  16457   617   
4         GD McGrath (AUS)  1993-2007  124   243  29248  12186   563   
5          SCJ Broad (ENG)  2007-2021  148   272  29713  14502   523   
6            CA Walsh (WI)  1984-2001  132   242  30019  12688   519   
7            DW Steyn (SA)  2004-2019   93   171  18608  10077   439   
8      N Kapil Dev (INDIA)  1978-1994  131   227  27740  12867   434   
9        HMRKB Herath (SL)  1999-2018   93   170  25993  12157   433   

                   BBI                  BBM    Ave  Econ    SR   5  10  
0  1951-09-01 00:00:00               16/220  22.72  2.47  55.0  67  22  
1  1971-08-01 00:00:00               12/128  25.41  2.65  57.4  37  10  
2  1974-10-01 00:00:00               14/149  29.65  2.69  65.9  35   8  
3  1942-07-01 00:00:00  1971-11-01 00:00:00  26.67  2.83  56.3  30   3  
4  2021-08-24 00:00:00  2021-10-27 00:00:00  21.64  2.49  51.9  29   3  
5  2021-08-15 00:00:00               11/121  27.72  2.92  56.8  18   3  
6  1937-07-01 00:00:00                13/55  24.44  2.53  57.8  22   3  
7  1951-07-01 00:00:00  1960-11-01 00:00:00  22.95  3.24  42.3  26   5  
8  1983-09-01 00:00:00               11/146  29.64  2.78  63.9  23   2  
9                9/127               14/184  28.07  2.80  60.0  34   9

### Checking number of rows and columns in data set

### Description of the features :

<br><strong>Player: </strong> Name of the players who played test cricket
<br><strong>Span: </strong> Career span of each player
<br><strong>Mat: </strong> Number of matches played
<br><strong>Inns: </strong> Total innings played by the player
<br><strong>Balls: </strong> Total number of balls bowled by the player
<br><strong>Runs: </strong> The number of runs conceded by the player
<br><strong>Wkts: </strong> Total number of wickets taken by the player
<br><strong>BBI: </strong> Best bowling in an innings of the player
<br><strong>BBM: </strong> Best bowling in a match of the player
<br><strong>Ave: </strong> The average number of runs conceded for per wicket by the player. (Runs/W)
<br><strong>Econ: </strong> The average number of runs conceded per over. (Runs/Overs bowled)
<br><strong>SR: </strong> The average number of balls bowled for per wicket taken. (Balls/W)
<br><strong>5: </strong> The number of innings in which the bowler took at least five wickets.
<br><strong>10: </strong> The number of matches in which the bowler took at least ten wickets.

In [5]:
# number of rows
print("number of rows = ", df.shape[0])

# number of columns
print("number of columns = ", df.shape[1])

number of rows =  79
number of columns =  14


### Features in the dataset and search for missing values and data types

In [11]:
display(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 79 entries, 0 to 78
Data columns (total 14 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Player                79 non-null     object 
 1   Carrer_Duration       79 non-null     object 
 2   Match                 79 non-null     int64  
 3   Innings               79 non-null     int64  
 4   Balls                 79 non-null     int64  
 5   Runs                  79 non-null     int64  
 6   Wickets               79 non-null     int64  
 7   Best_Bowling_Innings  79 non-null     object 
 8   Best_Bowling_Match    79 non-null     object 
 9   Bowling_Average       79 non-null     float64
 10  Economy_Rate          79 non-null     float64
 11  Strike_Rate           79 non-null     float64
 12  5                     79 non-null     int64  
 13  10                    79 non-null     int64  
dtypes: float64(3), int64(7), object(4)
memory usage: 8.8+ KB


None

##### Observations:

There are
> - 3 float type variables- Bowling average, Economy rate, Strike rate<br>
> - 4 String type variables- Best Bowling in Innings, Best Bowling in Match, Player, Span<br>
> - 7 integer type variables- Match, Innings, Balls, Runs, Wickets, 5 wickets in and innings, 10 wickets in an innings

> From the non-null columns, we can see that there are **no missing value** as of now.<br>

### Staistical Analysis of the dataset

In [6]:
display(df.describe())

Mat        Inns         Balls          Runs        Wkts  \
count   79.000000   79.000000     79.000000     79.000000   79.000000   
mean    80.101266  144.797468  18630.303797   8595.506329  317.101266   
std     28.537692   51.042310   7190.036515   3080.256645  121.731587   
min     37.000000   67.000000   8785.000000   4846.000000  200.000000   
25%     60.500000  110.000000  13580.000000   6456.500000  229.000000   
50%     71.000000  129.000000  16498.000000   7742.000000  266.000000   
75%     93.000000  169.000000  21742.500000   9756.000000  374.500000   
max    166.000000  301.000000  44039.000000  18355.000000  800.000000   

             Ave       Econ         SR          5         10  
count  79.000000  79.000000  79.000000  79.000000  79.000000  
mean   27.466456   2.806582  59.187342  16.354430   2.797468  
std     3.657561   0.351666   9.349337   9.642372   3.235935  
min    20.940000   1.980000  41.200000   3.000000   0.000000  
25%    24.425000   2.600000  53.300000   9.500000   1.000000  
50%    28.000000   2.820000  57.400000  14.000000   2.000000  
75%    29.870000   3.080000  63.950000  20.500000   3.500000  
max    34.790000   3.460000  91.900000  67.000000  22.000000

##### Statistic measures
- Count- total numbers of observation
- mean- average of the values
- std- standard deviation
- min- minimum value in the column
- 25%- 1st quartile or 25% of all value is under the certain column value
- 50%- 2nd quartile or 50% of all value is under the certain column value
- 75%- 3rd quartie or 75% of all value is under the certain column value
- max- maximum value in the column

##### Observations
> - On an average a played has played 144 innings
> - Minimum 67 innings were played by a player
> - 25% of players have played 110 innings or less
> - 50% of players have played 129 innings or less
> - 75% of players have played 169 innings or less
> - Maximum 301 innings were played by a player
> - Minimum 67 innings were played by a player

Many other primary insights can be taken from the statistics

### Renaming columns in dataset

In [8]:
#renameing columns in dataframe
df = df.rename(columns={'Span' : 'Carrer_Duration',
                        'Mat' : 'Match',
                        'Inns': 'Innings',
                        'Wkts' : 'Wickets',
                        'BBI' : 'Best_Bowling_Innings',
                        'BBM' : 'Best_Bowling_Match',
                        'Ave' : 'Bowling_Average',
                        'Econ': 'Economy_Rate',
                        'SR' : 'Strike_Rate'})
display(df.head())

Player Carrer_Duration  Match  Innings  Balls   Runs  \
0  M Muralitharan (ICC/SL)       1992-2010    133      230  44039  18180   
1           SK Warne (AUS)       1992-2007    145      273  40705  17995   
2         A Kumble (INDIA)       1990-2008    132      236  40850  18355   
3        JM Anderson (ENG)       2003-2021    162      301  34791  16457   
4         GD McGrath (AUS)       1993-2007    124      243  29248  12186   

   Wickets Best_Bowling_Innings   Best_Bowling_Match  Bowling_Average  \
0      800  1951-09-01 00:00:00               16/220            22.72   
1      708  1971-08-01 00:00:00               12/128            25.41   
2      619  1974-10-01 00:00:00               14/149            29.65   
3      617  1942-07-01 00:00:00  1971-11-01 00:00:00            26.67   
4      563  2021-08-24 00:00:00  2021-10-27 00:00:00            21.64   

   Economy_Rate  Strike_Rate   5  10  
0          2.47         55.0  67  22  
1          2.65         57.4  37  10  
2          2.69         65.9  35   8  
3          2.83         56.3  30   3  
4          2.49         51.9  29   3

### Removing columns from dataset

<br> The data in the column BBI and BBM was bugged and some of the entries got turned into datetime
<br> removing Best_Bowling_Innings and Best_Bowling_Match

In [14]:
df.drop(['Best_Bowling_Innings' , 'Best_Bowling_Match'] , axis=1 , inplace= True )

display(df.head())

Player Carrer_Duration  Match  Innings  Balls   Runs  \
0  M Muralitharan (ICC/SL)       1992-2010    133      230  44039  18180   
1           SK Warne (AUS)       1992-2007    145      273  40705  17995   
2         A Kumble (INDIA)       1990-2008    132      236  40850  18355   
3        JM Anderson (ENG)       2003-2021    162      301  34791  16457   
4         GD McGrath (AUS)       1993-2007    124      243  29248  12186   

   Wickets  Bowling_Average  Economy_Rate  Strike_Rate   5  10  
0      800            22.72          2.47         55.0  67  22  
1      708            25.41          2.65         57.4  37  10  
2      619            29.65          2.69         65.9  35   8  
3      617            26.67          2.83         56.3  30   3  
4      563            21.64          2.49         51.9  29   3